model/jpa/src/main/resources/META-INF/orm.xml

```diff
@@ -48,7 +48,8 @@ http://java.sun.com/xml/ns/persistence/orm_2_0.xsd"
             </many-to-one>
             <many-to-many name="categories" fetch="EAGER">
                 <cascade>
-                    <cascade-all/>
+                    <cascade-persist/>
+                    <cascade-merge/>
                 </cascade>
```

Comment:
You can't remove the categories that still have relations, this failed with the improved tests